sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -534,10 +538,20 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
             if child.resolved && AliasResolution.hasUnresolvedAlias(projectList) =>
           Project(AliasResolution.assignAliases(projectList), child)
 
+        case p @ Project(projectList, child) if p.containsPattern(PIPE_EXPRESSION) =>
```

Comment:
It's a bit tricky that we validate `PipeExpression` in the rule `ResolveAliases`. How about this:
1. Mark `PipeExpression` as unresolved, so that `ResolveAliases` will skip `PipeExpression` first.
2. Add a new rule to validate and strip `PipeExpression`.
3. After `PipeExpression` is stripped, `ResolveAliases` will be triggered normally.

Reply:
Sure, sounds good, this is done.